docs/storage-architecture.md

```diff
@@ -0,0 +1,189 @@
+
+**Responsibilities:**
+- Store and retrieve media file records
+- Track file metadata (hash, mime type, dimensions, blurhash)
+- Update accessed_at timestamps
+- Query files by group and hash
+
+**Does NOT:**
+- Touch the filesystem
+- Download or upload files
+- Handle encryption/decryption
+
+**Key Types:**
+- `MediaFile` - Database record representation
+- `MediaFileParams` - Parameters for saving records
+- `MediaFileRow` - Internal database row type
+
+### 3. `storage/media_files.rs` (Middle Branch)
+**Purpose:** Filesystem operations for media files
+
+**Responsibilities:**
+- Create directory structures (`<cache_dir>/<group_id_hex>/<subdirectory>/`)
+- Store files atomically (write to `.tmp`, then rename)
+- Retrieve file paths
+- Check file existence
+- Find files by prefix (useful when extension is unknown)
+
+**Does NOT:**
+- Interact with the database
+- Download or upload files
+- Handle encryption/decryption
+- Know about Blossom URLs
+
+**Key Types:**
+- `MediaFileStorage` - Filesystem storage manager
+- `Storage` - Wrapper struct for all storage subsystems
+
+**Directory Structure:**
+```
+<data_dir>/media_cache/
+  <mls_group_id_hex>/
+    group_images/          # Group profile images (marmot extension)
+      <hash>.<ext>
+    media/                 # Chat media files
+      <hash>.<ext>
+```
+
+### 4. `BlossomClient` (Right Branch)
+**Purpose:** Network operations with Blossom servers
+
+**Provided by:** `nostr_sdk` crate
+
+**Responsibilities:**
+- Upload blobs to Blossom servers
+- Download blobs from Blossom servers
+- Handle authentication with Nostr keys
+
+**Does NOT:**
+- Touch the filesystem directly
+- Manage database records
+- Handle caching
+
+## Usage Patterns
+
+### Group Images (Marmot Data Extension)
+
+Group images are part of the Marmot protocol's data extension and are stored in the `group_images` subdirectory.
+
+**Using the orchestration layer (recommended):**
+```rust
+// Store and record in one operation
+let path = whitenoise
+    .media_files()
+    .store_and_record(
+        &account_pubkey,
+        &group_id,
+        "group_images",
+        "abc123.jpg",
+        &decrypted_data,
+        &hash,
+        "image/jpeg",
+        "group_image",
+        Some("https://..."),
+        None,
+        None,
+    )
+    .await?;
+```
+
+**Using layers directly (for more control):**
+```rust
+// Store file to filesystem
+let path = whitenoise.storage.media_files
+    .store_file(&group_id, "group_images", "abc123.jpg", &decrypted_data)
+    .await?;
+
+// Record in database
+use whitenoise::database::media_files::{MediaFile, MediaFileParams};
+MediaFile::save(
+    &whitenoise.database,
+    &group_id,
+    &account_pubkey,
+    MediaFileParams {
+        file_path: &path,
+        file_hash: &hash,
+        mime_type: "image/jpeg",
+        media_type: "group_image",
+        blossom_url: Some("https://..."),
+        dimensions: None,
+        blurhash: None,
+    },
+).await?;
+```
```

Comment:
I'd either remove this or explicitly say this is intended to be used only on `whitenoise/media_files.rs`. I can foresee there being issues with db/storage syncing if we're not careful about how we're handling things